### ISSUE TITLE
About few shot question?

### ISSUE THREAD

**Issue by haiduo** [Nicolas-BZRD/llm-recipes#2]
Hi Nicolas-BZRD,

I still have a remaining question, why does the dataset SQuAD have few-shot=1? When I try to generate data according to shot=1 by 'generator.py', I find that the content of the 'context' included 'question' and 'answer'. Will training with such data be over-fitting?   
![image](https://github.com/Nicolas-BZRD/llm-recipes/assets/35062202/019a9dc7-ed5a-42cd-924f-f705ac345452)
So, I also try shot=0 and find that it can train and converge normally, as follows:  
https://api.wandb.ai/links/haiduo/m8q3ipz0  
In addition, I turned on bf16, and it seems that the size of the GPU memory is not much different from not turning on. I don’t know if I haven’t set it correctly.  
![image](https://github.com/Nicolas-BZRD/llm-recipes/assets/35062202/58837bd9-f727-416e-b9c4-85fbef4b306f)

**Comment by Nicolas-BZRD** [comment]
I don't understand. The code you provided is from the loading file. If you generate a dataset with a teacher in a few-shot setting, when you train the student model, the teacher needs the exact same few-shot examples used previously to regenerate the exact same answers with the same logits in order train the student.

For the second question, it's similar. If the dataset is generated in a few-shot setting (for example 1) and you change this value to 0 during the training, the teacher will not regenerate the same sentences. Consequently, the distilled logits will not be the logit corresponding to the answer stored in the dataset. This results in the student being asked to match some random logits. By introducing this randomness, you prevent the student from overfitting, but this may also lead to suboptimal performance.

Concerning bf16, all GPU aren't compatible, if it's your case bf16 is automatically turn off. Additionally, only the teacher is loaded in bf16 not the student.

I hope this will help you 🙌

**Comment by haiduo** [comment]
Sorry, I misunderstand the case of few-shot=1, adding a single example is always certain. Now, I understand the meaning of few-shot=1. For BF16, I am using an Nvidia A40. Although it shows that it supports the BF16 format, maybe I didn't set it correctly. I will carefully check the relevant settings.

Sincerely, thank you for your prompt answer.
